src/modules/blockchain/implementation/web3-service.js

```diff
@@ -75,6 +108,14 @@ class Web3Service {
         return this.config.identity;
     }
 
+    getBlockNumber() {
+        return this.web3.eth.getBlockNumber();
+    }
+
+    getBlockTime() {
+        return this.config.blockTime;
```

Comment:
Cant we somehow get this information from the chain?

Reply:
Probably we can, I will take a look